asv_bench/benchmarks/period.py

```diff
@@ -127,17 +54,78 @@ def time_start_time(self):
     def time_end_time(self):
         self.per.end_time
 
-    def time_to_timestamp():
+
+class PeriodUnaryMethods(object):
+    params = ['M', 'min']
+    param_names = ['freq']
+
+    def setup(self, freq):
+        self.per = Period('2012-06-01', freq=freq)
+
+    def time_to_timestamp(self):
         self.per.to_timestamp()
 
-    def time_now():
+    def time_now(self):
         self.per.now()
 
-    def time_asfreq():
+    def time_asfreq(self):
         self.per.asfreq('A')
 
 
-class period_standard_indexing(object):
+class PeriodIndexConstructor(object):
+    goal_time = 0.2
+
+    params = ['D']
+    param_names = ['freq']
+
+    def setup(self, freq):
+        self.freq = freq
+        self.rng = date_range('1985', periods=1000)
+        self.rng2 = date_range('1985', periods=1000).to_pydatetime()
+
+    def time_from_date_range(self):
```

Comment:
you also need to pass the param in the actual `time_` functions as well. So `time_from_date_range(self, freq)` in this case (and also for all other benchmarks)

Reply:
Is this in addition to or instead of setting `self.freq=freq` in `setup`?

Reply:
that doesn't matter, it is just needed to let the code actually run. 
But since you have to do that, the `self.freq = freq` is a bit superfluous as you can also do `PeriodIndex(.. freq=freq)`. So to conclude: instead